packages/translation-extension/src/index.ts

```diff
@@ -149,10 +149,11 @@ const langMenu: JupyterFrontEndPlugin<void> = {
 
         let command: string;
 
-        const serverSettings = app.serviceManager.serverSettings;
         // Get list of available locales
-        requestTranslationsAPI<any>('', '', {}, serverSettings)
-          .then(data => {
+        translatorConnector
+          .fetch({ language: '' })
+          // TODO: fix usage of any
```

Comment:
In its current state, the translation extension does not seem to be using proper types in the code base.

Added a `TODO` here so we can track improving that in an issue and come back to it later.

Reply:
It means `fetch` return type is incorrect. That means we are exposing an invalid contract in newly fixed `ITranslatorConnector`, so yeah this should be addressed before we release 4.4.

Maybe you could cherry-pick from my https://github.com/jupyterlab/jupyterlab/pull/16219 - it is rather large so pasting snapshot of relevant section:

![image](https://github.com/user-attachments/assets/2902f75b-37a4-4ea8-8c7e-1113122d020d)

And possibly also

![image](https://github.com/user-attachments/assets/eb1ff856-33df-47c5-89e3-7619d6fff792)

If you think it should be done in a separate PR that's fine too.

Reply:
> Maybe you could cherry-pick from my #16219 - it is rather large so pasting snapshot of relevant section:

Ah nice thanks for the pointer 👍 We should also try to get such PR across the finish line as it would be quite helpful in general.



> If you think it should be done in a separate PR that's fine too.

Yes happy to track separately (as a release blocker) and be assigned to it. This can be addressed later during the beta or rc cycles.

Reply:
>  We should also try to get such PR across the finish line as it would be quite helpful in general.

Yes, probably early in the 4.5 cycle.